Peridigm does not compile with gcc 8 due to more strict evaluation of the comparator

```
In file included from /usr/include/c++/8/map:60,
                 from /usr/local/trilinos/include/Teuchos_map.hpp:51,
                 from /usr/local/trilinos/include/Teuchos_RCPNode.hpp:54,
                 from /usr/local/trilinos/include/Teuchos_RCPDecl.hpp:51,
                 from /usr/local/trilinos/include/Teuchos_RCP.hpp:58,
                 from /peridigm/src/core/Peridigm_Enums.hpp:7,
                 from /peridigm/src/core/Peridigm_Enums.cpp:48:
/usr/include/c++/8/bits/stl_tree.h: In instantiation of 'class std::_Rb_tree<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, Tensor_Order>, std::_Select1st<std::pair<const std::__cxx11::basic_string<char>, Tensor_Order> >, std::__cxx11::basic_string<char>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, Tensor_Order> > >':
/usr/include/c++/8/bits/stl_map.h:151:17:   required from 'class std::map<std::__cxx11::basic_string<char>, Tensor_Order, std::__cxx11::basic_string<char> >'
/peridigm/src/core/Peridigm_Enums.cpp:160:50:   required from here
/usr/include/c++/8/bits/stl_tree.h:452:21: error: static assertion failed: comparison object must be invocable with two arguments of key type
       static_assert(__is_invocable<_Compare&, const _Key&, const _Key&>{},
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/8/bits/stl_tree.h: In instantiation of 'class std::_Rb_tree<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, Spatial_Coordinate>, std::_Select1st<std::pair<const std::__cxx11::basic_string<char>, Spatial_Coordinate> >, std::__cxx11::basic_string<char>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, Spatial_Coordinate> > >':
/usr/include/c++/8/bits/stl_map.h:151:17:   required from 'class std::map<std::__cxx11::basic_string<char>, Spatial_Coordinate, std::__cxx11::basic_string<char> >'
/peridigm/src/core/Peridigm_Enums.cpp:222:56:   required from here
/usr/include/c++/8/bits/stl_tree.h:452:21: error: static assertion failed: comparison object must be invocable with two arguments of key type
/usr/include/c++/8/bits/stl_tree.h: In instantiation of 'class std::_Rb_tree<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, Boundary_Condition_Type>, std::_Select1st<std::pair<const std::__cxx11::basic_string<char>, Boundary_Condition_Type> >, std::__cxx11::basic_string<char>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, Boundary_Condition_Type> > >':
/usr/include/c++/8/bits/stl_map.h:151:17:   required from 'class std::map<std::__cxx11::basic_string<char>, Boundary_Condition_Type, std::__cxx11::basic_string<char> >'
/peridigm/src/core/Peridigm_Enums.cpp:283:61:   required from here
/usr/include/c++/8/bits/stl_tree.h:452:21: error: static assertion failed: comparison object must be invocable with two arguments of key type
```

--- a/src/core/Peridigm_Enums.cpp
+++ b/src/core/Peridigm_Enums.cpp
@@ -157,13 +157,13 @@ Tensor_Order to_tensor_order(const std::string & str)
   std::string upper_str = str;
   tidy_string(upper_str);
 
-  std::map<std::string,Tensor_Order,std::string >::iterator pos=string_tensor_order.find(upper_str);
+  std::map<std::string,Tensor_Order>::iterator pos=string_tensor_order.find(upper_str);
   if (pos == string_tensor_order.end())
   {
     std::stringstream oss;
     oss << "ERROR: Unknown tensor order: " << upper_str << std::endl;
     oss << "Valid options include: " << std::endl;
-    for (std::map<std::string,Tensor_Order,std::string >::iterator it = string_tensor_order.begin(); it != string_tensor_order.end(); ++it)
+    for (std::map<std::string,Tensor_Order>::iterator it = string_tensor_order.begin(); it != string_tensor_order.end(); ++it)
     {
       oss << "  " << it->first << std::endl;
     }
@@ -219,13 +219,13 @@ Spatial_Coordinate to_spatial_coordinate(const std::string & str)
   std::string upper_str = str;
   tidy_string(upper_str);
 
-  std::map<std::string,Spatial_Coordinate,std::string >::iterator pos=string_spatial_coordinate.find(upper_str);
+  std::map<std::string,Spatial_Coordinate>::iterator pos=string_spatial_coordinate.find(upper_str);
   if (pos == string_spatial_coordinate.end())
   {
     std::stringstream oss;
     oss << "ERROR: Unknown spatial coordinate: " << upper_str << std::endl;
     oss << "Valid options include: " << std::endl;
-    for (std::map<std::string,Spatial_Coordinate,std::string >::iterator it = string_spatial_coordinate.begin(); it != string_spatial_coordinate.end(); ++it)
+    for (std::map<std::string,Spatial_Coordinate>::iterator it = string_spatial_coordinate.begin(); it != string_spatial_coordinate.end(); ++it)
     {
       oss << "  " << it->first << std::endl;
     }
@@ -280,13 +280,13 @@ Boundary_Condition_Type to_boundary_condition_type(const std::string & str)
   std::string upper_str = str;
   tidy_string(upper_str);
 
-  std::map<std::string,Boundary_Condition_Type,std::string >::iterator pos=string_boundary_condition.find(upper_str);
+  std::map<std::string,Boundary_Condition_Type>::iterator pos=string_boundary_condition.find(upper_str);
   if (pos == string_boundary_condition.end())
   {
     std::stringstream oss;
     oss << "ERROR: Unknown boundary condition type: " << upper_str << std::endl;
     oss << "Valid options include: " << std::endl;
-    for (std::map<std::string,Boundary_Condition_Type,std::string >::iterator it = string_boundary_condition.begin(); it != string_boundary_condition.end(); ++it)
+    for (std::map<std::string,Boundary_Condition_Type>::iterator it = string_boundary_condition.begin(); it != string_boundary_condition.end(); ++it)
     {
       oss << "  " << it->first << std::endl;
     }

